app/src/ui/branches/ci-status.tsx

```diff
@@ -36,11 +40,15 @@ export class CIStatus extends React.PureComponent<
 
   public constructor(props: ICIStatusProps) {
     super(props)
+    const check = props.dispatcher.tryGetCommitStatus(
+      this.props.repository,
+      this.props.commitRef
+    )
     this.state = {
-      check: props.dispatcher.tryGetCommitStatus(
-        this.props.repository,
-        this.props.commitRef
-      ),
+      check,
+    }
+    if (this.props.onCheckChange !== undefined) {
+      this.props.onCheckChange(check)
     }
```

Comment:
I think we could use this to clean up those `!== undefined` checks. What do you think?
```suggestion
    this.props.onCheckChange?.(check)
```

Reply:
I am totally good with that.. but I thought we were avoiding non-direct checks. :/

Reply:
What do you mean? 😳